invisible_cities/reco/xy_algorithms.py

```diff
@@ -170,7 +170,7 @@ def corona(pos, qs,
             masked_sipm = []
         # find any masked channel in the first ring around the new local maximum of charge
         neighbours = get_neighbours(new_local_maximum, pitch)
-        n_masked_neighbours = sum(1 for p in neighbours if len(list(filter(lambda x: np.allclose(p, x), masked_sipm))) == 1)
+        n_masked_neighbours = sum(len(list(filter(lambda x: np.allclose(p, x), masked_sipm))) for p in neighbours)
```

Comment:
Wrapping the filter in a list so that you can take its `len` in order to count the number of times a condition is satisfied, is probably not the clearest way of expressing the intent. So I would replace

    len(list(filter(lambda x: np.allclose(p,x), masked_sipm))

with

    sum(1 for x in masked_sipm if np.allclose(p,x))

and give `x` a meaningful name. But it would probably be even better to wrap this whole expression inside a function with a meaningful name, and call the function. Remember that such a function can be defined *inside* `corona` itself.

Try to make the code-self documenting: meaningful names attached to small functions or values of expressions often help to clarify what is going on, and why.